src/profile-logic/profile-data.js

```diff
@@ -439,21 +451,29 @@ export function getTimingsForCallNodeIndex(
     }
     timings.breakdownByImplementation[implementation] += duration;
 
-    // step 4: find the category value for this stack
-    const categoryIndex = stackTable.category[stackIndex];
-    const subcategoryIndex = stackTable.subcategory[stackIndex];
-
-    // step 5: increment the right value in the category breakdown
-    if (timings.breakdownByCategory === null) {
-      timings.breakdownByCategory = categories.map(category => ({
-        entireCategoryValue: 0,
-        subcategoryBreakdown: Array(category.subcategories.length).fill(0),
-      }));
+    // step 4: find the category value for this stack. We want to use the
+    // category of the unfilteredThread.
+    const unfilteredStackIndex =
+      unfilteredSamples.stack[sampleIndex + sampleIndexOffset];
+    if (unfilteredStackIndex !== null) {
+      const categoryIndex = unfilteredStackTable.category[unfilteredStackIndex];
+      const subcategoryIndex =
+        unfilteredStackTable.subcategory[unfilteredStackIndex];
```

Comment:
The lines 454 to 461 are the heart of this change.

I didn't do it for the breakdown by implementation because it's still super difficult to get that breakdown right for the combined view, so it's better to use the filtered thread instead. We'll still need a strategy for that issue.